### PR TITLE
CDAP-19589 - Use sort keys for reliable ordering of DML events for unordered streams

### PR DESCRIPTION
**Pull request by sumitjnn**
- Added tests (will be adding few more scenarios)
- Validated there are no checkstyle errors

### CODE REVIEW

File: src/main/java/io/cdap/delta/bigquery/BigQueryTableState.java [hunk 1]
```diff
@@ -28,14 +33,31 @@ public class BigQueryTableState {
 
   private final List<String> primaryKeys;
 
+  @Nullable
+  private List<Schema.Type> sortKeys;
+
   public BigQueryTableState(List<String> primaryKeys) {
     this.primaryKeys = Collections.unmodifiableList(new ArrayList<>(primaryKeys));
```
Comment (seanzhougoogle) [inline src/main/java/io/cdap/delta/bigquery/BigQueryTableState.java]: nit: call the other constructor and pass osrtKeys as empty list

File: src/main/java/io/cdap/delta/bigquery/BigQueryEventConsumer.java [hunk 2]
```diff
@@ -493,15 +500,14 @@ private void updatePrimaryKeys(TableId tableId, List<String> primaryKeys) throws
       return;
     }
     primaryKeyStore.put(tableId, primaryKeys);
-    context.putState(String.format("bigquery-%s-%s", tableId.getDataset(), tableId.getTable()),
+    context.putState(getTableStateKey(tableId),
                      Bytes.toBytes(GSON.toJson(new BigQueryTableState(primaryKeys))));
```
Comment (seanzhougoogle) [inline src/main/java/io/cdap/delta/bigquery/BigQueryEventConsumer.java]: when primary key is updated, we don't preserve the old sortkey in the table state ?

Comment (sumitjnn) [reply src/main/java/io/cdap/delta/bigquery/BigQueryEventConsumer.java]: Fixed

File: src/main/java/io/cdap/delta/bigquery/Constants.java [hunk 3]
```diff
@@ -26,6 +26,8 @@ public final class Constants {
   public static final String ROW_ID = "_row_id";
   public static final String OPERATION = "_op";
   public static final String BATCH_ID = "_batch_id";
+  public static final String SORT_KEYS = "_sort";
```
Comment (seanzhougoogle) [inline src/main/java/io/cdap/delta/bigquery/Constants.java:29]: nit: _sort -> _sort_keys

Comment (sumitjnn) [reply src/main/java/io/cdap/delta/bigquery/Constants.java:29]: I wanted to keep is the struct field as `_sort` and the actual nested key as `_key_0` (column - `_sort._key_0`), wdyt?

**Comment by sumitjnn** [comment]
@sahusanket @seanzhougoogle Could you please review the PR or approve if there are no further issues?

File: src/main/java/io/cdap/delta/bigquery/Constants.java (hunk 3 above)
Comment (seanzhougoogle) [reply src/main/java/io/cdap/delta/bigquery/Constants.java:29]: I see what you meant , sg.

**Review by seanzhougoogle — APPROVED** [review]

**Review by sahusanket — APPROVED** [review]

**Pull request merged**